aten/src/ATen/native/cuda/Sort.cu

```diff
@@ -266,6 +267,9 @@ std::tuple<Tensor &,Tensor &> sort_out_stable_cuda(const Tensor & self, c10::opt
     return std::forward_as_tuple(values, indices);
   }
 
+  TORCH_CHECK(get_overlap_status(self, values) == MemOverlapStatus::NO,
+    "Sort size too large for inplace sort");
```

Comment:
Size too large?

Reply:
What I mean is, you should tell the user that memory overlap is not support supported. Not size too large.

Reply:
Even better, for this case, when there is overlap, you should output to a temporary memory, and copy_ it back.

Reply:
Is it supported for smaller input? Or do we want to force non-overlapping regardless?

Reply:
Smaller inputs do inplace sort always (they start by copying to output if sort is not inplace)